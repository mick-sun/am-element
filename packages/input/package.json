{
  "name": "am-form-input",
  "version": "0.0.3",
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "license": "MIT",
  "scripts": {
    "del": "rimraf dist"
  },
  "dependencies": {
    "lodash": "3"
  },
  "gitHead": "4ea3a9eea9a1897b39f4a67b5a2a24860ed592b4"
}
