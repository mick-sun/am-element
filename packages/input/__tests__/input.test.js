import input from '../src/input'
import { mount } from '@vue/test-utils'

describe('am-form-input', () => {
  test('input-text', () => {
    const wrapper = mount(input)
    expect(wrapper.html()).toContain('input type="text"')
  })

  test('input-password', () => {
    const wrapper = mount(input, {
      propsData: {
        type: 'password'
      }
    })
    expect(wrapper.html()).toContain('input type="password"')
  })

  test('input-value', () => {
    const wrapper = mount(input, {
      propsData: {
        type: 'password',
        value: 'admin'
      }
    })
    expect(wrapper.props('value')).toBe('admin')
  })

  test('input-snapshot', () => {
    const wrapper = mount(input, {
      propsData: {
        type: 'text',
        value: 'admin'
      }
    })
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
