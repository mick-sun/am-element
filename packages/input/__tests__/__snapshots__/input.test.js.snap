// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`am-form-input input-snapshot 1`] = `
<div>
  <input
    type="text"
  />
</div>
`;
